components/camel-sip/src/main/java/org/apache/camel/component/sip/listener/SipPresenceAgentListener.java

```diff
@@ -179,6 +187,64 @@ public void processSubscribe(RequestEvent requestEvent,
         }
     }
 
+    public void processMessage(RequestEvent requestEvent, ServerTransaction serverTransactionId) {
+        Request request = requestEvent.getRequest();
+        try
+        {
+            LOG.debug("SipPresenceAgentListener: Received an MESSAGE request");
```

Comment:
@nikvaessen are you sure it is always ```MESSAGE``` at this point?  Maybe check the request method == ```MESSAGE``` and log an error if something else comes in.